src/ui/actions/app.ts

```diff
@@ -53,6 +55,64 @@ function now(): number {
   return Date.now();
 }
 
+export const refetchDataForTimeRange = (focusRegion: FocusRegion): UIThunkAction => {
+  return async (dispatch, getState, { ThreadFront }) => {
+    // Technically, we only need to do this in *some* circumstances.
+    // I think the cases where we need to are:
+    // - We are enlarging or moving the focus zone beyond what we have most recently loaded.
+    // - We are shrinking the focus zone and have overflow currently.
+    // That first bullet might sounds like we will end up doing this basically
+    // all the time, but that is not so. Take, for instance, the case where the
+    // initial load of the recording did *not* set the overflow flag. In that
+    // case, we have all console messages, and we never need to fetch them
+    // again. However, in order to track this properly, I suspect we will need
+    // to add another piece of state to our messages slice: the boundaries that
+    // were most recently used to load messages. By default, this will be the
+    // entire recording, even if we start with only a small region focused.
+    // That's just the way that the Console.findMessages protocol command works.
+    // In that case (without overflow), like I said above, we never need to run
+    // any of this code. If we have overflow, and focus on, let's say the first
+    // 30 seconds of the recording, and we *don't* get overflow on the first
+    // thirty seconds, then we know that if the user were to change the window
+    // to the first 15 seconds, we don't need to refetch, and we also should not
+    // update the "most recently loaded" boundaries in the store, because that
+    // way the user could then focus on, for example, seconds 10 to 20 *also
+    // without refetching*.
+    // BTW - we don't have to figure out *all* of this right now :)
+    dispatch(clearMessages());
+    const endpoint = (await sendMessage("Session.getEndpoint", {}, ThreadFront.sessionId!)).endpoint;
```

Comment:
:speech_balloon: two quick thoughts here:

- It feels like `endpoint` is something that could be fetched once, on app startup, and we wouldn't have to fetch it again here
- Can the `getPointNearTime` requests be made in parallel instead of serial?  Probably not a big deal in terms of execution time, but doesn't _seem_ like we'd need to wait for the first to complete before requesting the second